File: backend/routes/workspace.go
package routes

import (
	"app/config"
	"app/controllers/workspace"

	"github.com/gin-gonic/gin"
)

func InitWorkspace(r *gin.RouterGroup, e *gin.Engine) {
	r.GET("/", workspace.Index)
	r.POST("/", workspace.Create)
	r.GET("/:id", workspace.Show)
	r.PUT("/:id", workspace.Update)
	r.DELETE("/:id", workspace.Delete)


	r.POST("/:id/traefik.yml", workspace.GenerateTraefikConfig)
	r.PUT("/:id/traefik.yml", workspace.UpdateTraefikConfig)

	e.GET(config.Config.Server.BaseUrl + "/workspaces/:id/traefik.yml", workspace.ShowTraefikConfig)
	e.GET(config.Config.Server.BaseUrl + "/workspaces/:id/version", workspace.ShowConfigVersion)
}

File: frontend/src/api/workspace.ts
import http from './_http';

export default {
  index () {
    return http.get('/workspaces/');
  },
  show (id: number) {
    return http.get(`/workspaces/${id}`);
  },
  create (data: any) {
    return http.post('/workspaces/', data);
  },
  update (id: number, data: any) {
    return http.put(`/workspaces/${id}`, data);
  },
  delete (id: number) {
    return http.delete(`/workspaces/${id}`);
  },
  generateTraefikConfigAsync (id: number) {
    return http.post(`/workspaces/${id}/traefik.yml`);
  },
  updateTraefikConfigAsync (id: number, yamlConfig: string, jsonConfig: string) {
    return http.put(`/workspaces/${id}/traefik.yml`, {traefik_config: yamlConfig, traefik_json_config: jsonConfig});
  },
}